src/metrics.ts

```diff
@@ -369,6 +380,12 @@ export default class Metrics extends EventEmitter {
       yggdrasilVersion: null,
       specVersion: SUPPORTED_SPEC_VERSION,
     };
+
+    if (impactMetrics.length > 0) {
+      (base as any).impactMetrics = impactMetrics;
```

Comment:
if we're doing type assertion maybe we can assert it's MetricsData & {impactMetrics: ImpactMetricsType}